### ISSUE TITLE
No such module "Firebase" in Swift Tests (Add explicit Firebase modulemap)

### ISSUE THREAD

**Issue by ryanwilson** [firebase/firebase-ios-sdk#58]
### [READ] Step 1: Are you in the right place?
Yes

### [REQUIRED] Step 2: Describe your environment

  * Xcode version: Any
  * Firebase SDK version: 4.0.0
  * Library version: 4.0.0
  * Firebase Product: All (auth, database, storage, core, messaging, etc)

### [REQUIRED] Step 3: Describe the problem

Swift unit tests fail to find Firebase module.  During the distribution process, the Firebase podspec relies on `Core` to set up proper header search paths using [`user_target_xcconfig`](https://github.com/CocoaPods/Specs/blob/master/Specs/0/3/5/Firebase/4.0.0/Firebase.podspec.json#L40-L42). Firebase should have its own `modulemap` instead of relying on the header search paths from Core.

#### Steps to reproduce:

1. Create a Swift project with unit tests
2. `pod init` to create a `Podfile`
3. Add `pod 'Firebase'` to `Podfile` under `use_frameworks!`
4. Run `pod update`
5. Open the generated xcworkspace
6. Add `import Firebase` to the `AppDelegate.swift`
7. Add `@testable import <YourProjectName>` into `<YourProjectName>Tests.swift`
8. A "No such module 'Firebase'" error will appear.

#### Relevant Code:

[Currently generated podspec](https://github.com/CocoaPods/Specs/blob/master/Specs/0/3/5/Firebase/4.0.0/Firebase.podspec.json#L40-L42)

#### Relevant Issues:
[Swift unit tests fail to find Firebase module](#16)
[Swift Cocoapod can't import Firebase pod dependencies](https://github.com/CocoaPods/CocoaPods/issues/6138)
[Unable to run tests (quickstarts)](https://github.com/firebase/quickstart-ios/issues/220)

#### Workarounds:

- Add `"${PODS_ROOT}/Firebase/Core/Sources"` to your **Tests target only** under `Build Settings -> Header Search Paths`

**Comment by morganchen12** [comment]
Is this not a duplicate of #16?

**Comment by zapjonny** [comment]
Any progress on this?

**Comment by zapjonny** [comment]
Btw the workaround does not work for me. 
CocoaPods notes that I am using these versions.

```
Using Firebase (3.17.0)
Using FirebaseAnalytics (3.9.0)
Using FirebaseAuth (3.1.1)
Using FirebaseCore (3.6.0)
Using FirebaseDatabase (3.1.2)
Using FirebaseInstanceID (1.0.10)
Using FirebaseMessaging (1.2.3)
Using FirebaseRemoteConfig (1.3.4)

```

I am not using Firebase 4.x but I assume this is not fixed yet?

**Comment by paulb777** [comment]
@zapjonny The issue has not yet been fixed, but if the workaround does not work for you, there is likely another issue with your configuration. Please create another issue with enough detail to reproduce the problem.

**Comment by zapjonny** [comment]
OK I just started unit testing as late as yesterday and probably failed my setup. Apparently you need to add your unit test target as a target in the podfile too (unless I got that wrong too). Anyway as soon as I did that I didn't have any problems running my unit tests.

Before that I manually installed the Firebase iOS SDK (upgrading to 4+), and I was able to work around the problem that way too.

**Comment by zapjonny** [comment]
And as far as I know, I am not using the suggested workaround from above but my unit tests still work. Maybe I did something different during setup.

**Comment by tirrorex** [comment]
FIY it is not failing just the tests, i have a swift framework using firebase and i cannot use it at the moment because firebase fails to import, still looking for a workaround

**Comment by paulb777** [comment]
@tirrorex You're running into a different issue: There is currently no clean way for frameworks to depend on Firebase. See https://github.com/CocoaPods/CocoaPods/issues/5368

**Comment by tirrorex** [comment]
i am already following this issue :/
Good to know this is separate but...

**Comment by ryanwilson** [comment]
Update:
- Still investigating looking at a fix we can provide on our side
- A bug was filed on the CocoaPods repo, "Search path not propagated to test target", https://github.com/CocoaPods/CocoaPods/issues/6891

**Comment by aadjemonkeyrock** [comment]
I've probably a related problem. I've a xcode project with multiple targets. They all use Firebase and the same set of other pods. It' all looks ok, but I can only run one (sort of random after install) of the targets. Switching between targets, cleaning and building will fail on the 'No such module Firebase'

I've tried the work arround, but that doesn't seem to work either.

**Comment by akmalshukri** [comment]
after do pod update then no such module firebase ? how to resolve this error ?

**Comment by paulb777** [comment]
This issue is fixed with CocoaPods 1.4.0.  For earlier CocoaPods releases, please use the workaround describe [here](https://github.com/firebase/firebase-ios-sdk/issues/58#issue-233253153).

If neither solution is working for you, please create another issue with full information about your configuration.
